webapp/src/Controller/API/LanguageController.php

```diff
@@ -122,6 +123,84 @@ public function updateExecutableActions(Request $request, string $id): void
         $this->dj->auditlog('executable', $language->getLangid(), 'updated');
     }
 
+    #[IsGranted('ROLE_ADMIN')]
+    #[Rest\Post('languages')]
+    #[OA\Response(
+        response: 200,
+        description: 'Configure all specified languages.',
+    )]
+    public function configureLanguagesAction(Request $request): Response
+    {
+        /** @var UploadedFile|null $jsonFile */
+        $jsonFile = $request->files->get('json');
+        if (!$jsonFile) {
+            throw new BadRequestHttpException('No JSON file supplied.');
+        }
+        $newLanguages = $this->dj->jsonDecode(file_get_contents($jsonFile->getRealPath()));
+
+        // Disable submission for all current languages, we will enable it for all new languages below.
+        $curLanguages = $this->em->getRepository(Language::class)->findAll();
+        foreach ($curLanguages as $language) {
+            /** @var Language $language */
+            $language->setAllowSubmit(false);
+        }
+
+        $idField = $this->eventLogService->externalIdFieldForEntity(Language::class) ?? 'langid';
+        foreach ($newLanguages as $language) {
+            /** @var Language $language */
+            $lang_id = $language['id'];
+            $lang = $this->em->getRepository(Language::class)->findOneBy(
+                [$idField => $lang_id]
+            );
+            if (!$lang) {
+                continue;
```

Comment:
Should we not indicate this to the one uploading that the language is missing?

Reply:
Yes, and no:

- Logging is not really possible as we return the updated languages config, so that it can be compared in the script.
- We could error out, but that's quite brutal.
- We could try adding a new language, but of course stuff like executables are missing here.

What about this: I will add a TODO comment here for now; and when we decided on the best solution we can replace it?

Reply:
I'm fine with that.